src/Tribe/Views/V2/iCalendar/Traits/Outlook_Methods.php

```diff
@@ -53,29 +64,69 @@ protected function generate_outlook_add_url_parameters( $calendar = 'live' ) {
 		// Getting the event details
 		$event = tribe_get_event();
 
-		$path = '/calendar/action/compose';
-		$rrv  = 'addevent';
-
 		/**
 		 * If event is an all day event, then adjust the end time.
 		 * Using the 'allday' parameter doesn't work well through time zones.
 		 */
 		if ( $event->all_day ) {
-			$enddt = Dates::build_date_object( $event->end_date )->format( 'Y-m-d' )
+			$end_datetime = Dates::build_date_object( $event->end_date )->format( 'Y-m-d' )
 				. 'T'
 				. Dates::build_date_object( $event->start_date )->format( 'H:i:s' );
 		} else {
-			$enddt = Dates::build_date_object( $event->end_date )->format( 'c' );
-			$enddt = substr( $enddt, 0, strlen( $enddt ) - 6 );
+			$end_datetime = Dates::build_date_object( $event->end_date )->format( 'c' );
+			$end_datetime = substr( $end_datetime, 0, -6 );
 		}
 
-		$startdt = Dates::build_date_object( $event->start_date )->format( 'c' );
-		$startdt = substr( $startdt, 0, strlen( $startdt ) - 6 );
+		$start_datetime = Dates::build_date_object( $event->start_date )->format( 'c' );
+		$start_datetime = substr( $start_datetime, 0, -6 );
+
+		$params = [
+			'path'     => '/calendar/action/compose',
+			'rrv'      => 'addevent',
+			'startdt'  => $start_datetime,
+			'enddt'    => $end_datetime,
+			'location' => Venue::generate_string_address( $event ),
+			'subject'  => $this->space_replace_and_encode( wp_strip_all_tags( $event->post_title ) ),
+			'body'     => $this->generate_outlook_event_description( $event ),
+		];
 
-		$location = Venue::generate_string_address( $event );
+		/**
+		 * Allow users to filter the params for our Outlook links before constructing the URL.
+		 *
+		 * @since TBD
+		 *
+		 * @var array    $params   The params used in the add_query_arg.
+		 * @var /WP_Post $event    The Event the link is for. As decorated by tribe_get_event().
+		 * @var string   $calendar The slug of the calendar. Values can be "outlook-365" and "outlook-live".
+		 */
+		$params = apply_filters( 'tec_events_single_event_outlook_link_parameters', $params, $event, $calendar );
 
-		$subject = $this->space_replace_and_encode( strip_tags( $event->post_title ) );
+		$service_slug = static::$service_slug;
```

Comment:
```suggestion
		$service_slug = static::get_service_slug();
```
Getter like we do elsewhere? Just a thought.